### PR TITLE
fs: add File::into_std and File::try_into_std methods

### PR DESCRIPTION
**Pull request by bartlomieju**
## Motivation

In version 0.1 there was File::into_std method that destructured
tokio_fs::File into std::fs:File. That method was lacking in
version 0.2.

Fixes: #1852

### CODE REVIEW

File: tokio/src/fs/file.rs [hunk 1]
```diff
@@ -394,6 +394,53 @@ impl File {
...
+    ///
+    /// # async fn dox() -> std::io::Result<()> {
+    /// let tokio_file = File::open("foo.txt").await?;
+    /// let std_file = file.into_std().await;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub async fn into_std(mut self) -> sys::File {
+        self.complete_inflight().await;
+        let std_file = Arc::downcast::<sys::File>(self.std).expect("Arc::downcast failed");
```
Comment (carllerche) [inline tokio/src/fs/file.rs]: Why is the downcast needed here?

File: tokio/src/fs/file.rs [hunk 2]
```diff
@@ -394,6 +394,53 @@ impl File {
...
+    /// ```no_run
+    /// use tokio::fs::File;
+    ///
+    /// # async fn dox() -> std::io::Result<()> {
+    /// let tokio_file = File::open("foo.txt").await?;
+    /// let std_file = file.try_into_std().await.unwrap();
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub async fn try_into_std(mut self) -> Result<sys::File, Self> {
```
Comment (carllerche) [inline tokio/src/fs/file.rs]: This does not need to be `async`.

File: tokio/src/fs/file.rs [hunk 3]
```diff
@@ -394,6 +394,53 @@ impl File {
         Ok(File::from_std(std_file))
     }
 
+    /// Destructures `File` into a [`std::fs::File`][std].
```
Comment (carllerche) [inline tokio/src/fs/file.rs]: Would you be able to add a short paragraph under this explaining that the function is async to allow for any in-flight operations to complete and to use `try_into_std` to attempt the conversion immediately.

File: tokio/src/fs/file.rs (hunk 2 above)
Comment (bartlomieju) [reply tokio/src/fs/file.rs]: Right :+1:

File: tokio/src/fs/file.rs (hunk 1 above)
Comment (bartlomieju) [reply tokio/src/fs/file.rs]: TBH I took your suggestion from the issue. I guess I can directly `Arc::try_unwrap(std_file).expect("Arc::try_unwrap failed")` after awaiting to complete inflight operations

**Comment by carllerche** [comment]
Thanks!

**Pull request merged**